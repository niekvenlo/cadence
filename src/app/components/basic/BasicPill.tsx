import styled from "styled-components";

export default styled.span`
  background: darkgreen;
  border-radius: 2em;
  color: white;
  font-size: 60%;
  margin: 0.5em;
  padding: 0.3em 1em;
  white-space: nowrap;
`;
