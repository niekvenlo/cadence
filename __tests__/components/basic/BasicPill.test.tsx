import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import BasicPill from "../../../src/app/components/basic/BasicPill";

describe("BasicPill", () => {
  it("renders a pill", () => {
    render(<BasicPill>label</BasicPill>);
    const button = screen.getByText("label");

    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle("background: darkgreen; white-space: nowrap");
    expect(button).toMatchSnapshot();
  });
});
