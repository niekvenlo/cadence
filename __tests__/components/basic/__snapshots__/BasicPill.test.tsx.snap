// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicPill renders a pill 1`] = `
<span
  class="sc-3e6d2b04-0 jMqrYW"
>
  label
</span>
`;
